spotify/models/user.py

```diff
@@ -28,6 +30,63 @@
 REFRESH_TOKEN_URL = "https://accounts.spotify.com/api/token?grant_type=refresh_token&refresh_token={refresh_token}"
 
 
+class TokenInfo:
+    def __init__(
```

Comment:
Allowing most of these arguments to be None will make it uncomfortable to use and too many edge cases appear.

Reply:
I was debating on what arguments should be mandatory. Having it this way allows to still use the flow of only providing an access token or only provide a refresh token.

Maybe it would make sense to always require access, refresh and expiry time when manually creating a user object to make the auth handling more consistent. This would break the existing `from_token` method though.